inc/namespace.php

```diff
@@ -456,3 +459,97 @@ function on_request_stats( TransferStats $stats ) {
 
 	on_aws_guzzle_request_stats( $stats );
 }
+
+/**
+ * Get the EC2 Instance metadata.
+ *
+ * This will cause a remote request to the metadata service when the cache is empty.
+ *
+ * @return array $array(
+ *    accountId: string,
+ *    architecture: string,
+ *    availabilityZone: string,
+ *    billingProducts: string,
+ *    devpayProductCodes: string,
+ *    marketplaceProductCodes: string,
+ *    imageId: string,
+ *    instanceId: string,
+ *    instanceType: string,
+ *    kernelId: string,
+ *    pendingTime: string,
+ *    privateIp: string,
+ *    ramdiskId: string,
+ *    region: string,
+ *    version: string,
+ * )
+ */
+function get_ec2_instance_metadata() : array {
+	$has_cache = false;
+	$cache_key = 'altis.ec2_instance_metadata';
+	// Use apcu_* as we only want to store the cache on the current server,
+	// not across all servers (wp_cache_*).
+	$cached_data = apcu_fetch( $cache_key, $has_cache );
+	if ( $has_cache ) {
+		return $cached_data;
+	}
+
+	$client = new Client();
+
+	try {
+		$request = $client->request( 'GET', 'http://169.254.169.254/latest/dynamic/instance-identity/document', [
```

Comment:
Any value in documenting where this IP comes from? Assuming it's a standard AWS thing.

Reply:
Yep, it's the standard link-local IP: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html

Reply:
IMO this IP is common enough I think.